### PR TITLE
Fix error management

### PR DESCRIPTION
**Pull request by LeoPelan**

### CODE REVIEW

File: src/Manager/MandateManager.php [hunk 1]
```diff
@@ -42,21 +39,22 @@ public function createMandateRequest(
...
         $mandateRequestData = [
             'id' => $data['Id'],
-            'BankAccount' => [
-                'Id' => $bankAccountId,
+            'bankAccount' => [
+                'id' => $bankAccountId,
+                'href' => $data['BankAccount']['Href'],
             ],
-            'date' => $data['Date'],
+            'date' => substr($data['Date'], 0, strrpos($data['Date'], '.')),
```
Comment (sylfabre) [inline src/Manager/MandateManager.php]: Should be a `DateTime`

File: src/Manager/MandateManager.php [hunk 2]
```diff
@@ -69,12 +67,10 @@ public function createMandateRequest(
      * Retrieve a mandate
      * @param  int $id
      * @param  string $appUserId
-     * @return MandateRequest
+     * @return array
```
Comment (sylfabre) [inline src/Manager/MandateManager.php]: Should be typed

File: src/Manager/MandateManager.php [hunk 3]
```diff
@@ -84,19 +80,25 @@ public function getMandate(string $appUserId, int $id): MandateRequest
...
+            'date' => $data['Date'],
             'status' => $data['Status'],
             'UMR' => $data['UMR'],
-            'date' => $data['Date'],
         ];
 
-        return new MandateRequest($mandateData);
+        return [
+            $mandateData,
+            'mandateDemands' => isset($data['mandateDemands']) ? $data['mandateDemands'] : null,
```
Comment (sylfabre) [inline src/Manager/MandateManager.php]: an empty array is `[]`, not `null`

File: src/Object/MandateRequest.php [hunk 4]
```diff
@@ -39,11 +39,10 @@ class MandateRequest extends AbstractHydratable
      * MandateRequest Status
      *
      * Values :
-     * 0 = Incomplete
-     * 1 = Waiting
-     * 2 = Refused
-     * 3 = Accepted
-     *
+     * 0 = Pending
```
Comment (sylfabre) [inline src/Object/MandateRequest.php]: define public constants

File: tests/Manager/MandateManagerTest.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            'http://test.com/returnurl/',
+            'http://services.assoconnect-dev.com/services/smoney/callback'
+        );
+
+        $file1 = __DIR__ . '/../data/sample.pdf';
+        $stream1 = fopen($file1, 'r+');
+        $filesize1 = filesize($file1);
+        $file1 = new UploadedFile($stream1, $filesize1, UPLOAD_ERR_OK, 'document.png', 'image/png');
+
+
```
Comment (sylfabre) [inline tests/Manager/MandateManagerTest.php]: double blank line

File: tests/Manager/MandateManagerTest.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        $bankAccountManager = new BankAccountManager($client);
+        $bankAccount = $bankAccountManager->createBankAccount($userPro, $bankAccountTest);
+
+        $mandateManager = new MandateManager($client);
+
+        $mandateRequest = $mandateManager->createMandateRequest(
+            $userPro->appUserId,
+            $bankAccount->id,
+            'http://test.com/returnurl/',
+            'http://services.assoconnect-dev.com/services/smoney/callback'
```
Comment (sylfabre) [inline tests/Manager/MandateManagerTest.php]: this exposes our internal logic

File: tests/Manager/MandateManagerTest.php [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        $filesize1 = filesize($file1);
+        $file1 = new UploadedFile($stream1, $filesize1, UPLOAD_ERR_OK, 'document.png', 'image/png');
+
+
+        $mandate = $mandateManager->getMandate(
+            $userPro->appUserId,
+            $mandateRequest->id
+        );
+
+        $paperMandateRequest = $mandateManager->sendPaperMandate(
```
Comment (sylfabre) [inline tests/Manager/MandateManagerTest.php]: `$paperMandateRequest` is a boolean but its name suggests something else

File: src/Manager/MandateManager.php (hunk 2 above)
Comment (LeoPelan) [reply src/Manager/MandateManager.php]: How ? 
We return an array with object and potential errors.

File: tests/Manager/MandateManagerTest.php (hunk 6 above)
Comment (LeoPelan) [reply tests/Manager/MandateManagerTest.php]: There isn't any test variable the 'callback-url'. Should I drop this test ?

**Pull request merged**